src/status_im/chat/handlers.cljs

```diff
@@ -530,3 +530,38 @@
       (if (= network-status :offline)
         (chats/inc-message-overhead chat-id)
         (chats/reset-message-overhead chat-id)))))
+
+(reg-fx
+  ::save-public-chat
+  (fn [chat]
+    (chats/save chat)))
+
+(reg-fx
+  ::start-watching-group
+  (fn [{:keys [group-id web3 current-public-key keypair]}]
+    (protocol/start-watching-group!
+      {:web3     web3
+       :group-id group-id
+       :identity current-public-key
+       :keypair  keypair
+       :callback #(dispatch [:incoming-message %1 %2])})))
+
+(register-handler-fx
+  :create-new-public-chat
+  (fn [{:keys [db]} [_ topic]]
+    (let [exists? (boolean (get-in db [:chats topic]))
+          chat    {:chat-id     topic
+                   :name        topic
+                   :color       default-chat-color
+                   :group-chat  true
+                   :public?     true
+                   :is-active   true
+                   :timestamp   (random/timestamp)}]
+      (merge
```

Comment:
so i need to return `{:a :b :c :d}` or `{:d}` i'm using `(merge (when condition
 {:a :b :c}) {:d}` how can i do it using `assoc` without using `let`?